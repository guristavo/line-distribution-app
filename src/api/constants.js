export const ALTERNATIVE_COLOR_LIST = {
  1: [31, 4, 17],
  2: [32, 5, 18],
  3: [33, 6, 19],
  4: [1, 7, 20],
  5: [2, 8, 21],
  6: [3, 9, 22],
  7: [4, 10, 23],
  8: [5, 11, 24],
  9: [6, 12, 25],
  10: [7, 13, 26],
  11: [8, 14, 27],
  12: [9, 15, 28],
  13: [10, 16, 29],
  14: [11, 17, 30],
  15: [12, 18, 31],
  16: [13, 19, 32],
  17: [14, 20, 33],
  18: [15, 21, 1],
  19: [16, 22, 2],
  20: [17, 23, 3],
  21: [18, 24, 4],
  22: [19, 25, 5],
  23: [20, 26, 6],
  24: [21, 27, 7],
  25: [22, 28, 8],
  26: [23, 29, 9],
  27: [24, 30, 10],
  28: [25, 31, 11],
  29: [26, 32, 12],
  30: [27, 33, 13],
  31: [28, 1, 14],
  32: [29, 2, 15],
  33: [30, 3, 16],
  34: [3, 8, 13, 26, 32],
  35: [4, 7, 12, 27],
  36: [6, 9, 16, 22, 31],
};
