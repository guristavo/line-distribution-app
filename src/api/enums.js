export const GENDERS = {
  FEMALE: 'Female',
  MALE: 'Male',
  UNKOWN: 'Unknown',
};

export const GENRES = {
  CPOP: 'C-Pop',
  JPOP: 'J-Pop',
  KPOP: 'K-Pop',
  OTHER: 'Other',
  POP: 'Pop',
};

export const NATIONALITIES = {
  AMERICAN: 'American',
  BRAZILIAN: 'Brazilian',
  BRITISH: 'British',
  CANADIAN: 'Canadian',
  CHINESE: 'Chinese',
  JAPANESE: 'Japanese',
  KOREAN: 'Korean',
  OTHER: 'Other',
  TAIWANESE: 'Taiwanese',
  THAI: 'Thai',
  VIETNAMISE: 'Vietnamise',
};

export const POSITIONS = {
  LEADER: 'Leader',
  MAIN_VOCALIST: 'Main Vocalist',
  MAIN_RAPPER: 'Main Rapper',
  MAIN_DANCER: 'Main Dancer',
  LEAD_VOCALIST: 'Lead Vocalist',
  LEAD_RAPPER: 'Lead Rapper',
  LEAD_DANCER: 'Lead Dancer',
  VOCALIST: 'Vocalist',
  RAPPER: 'Rapper',
  DANCER: 'Dancer',
  CENTER: 'Center',
  FACE: 'Face of the Group',
  VISUAL: 'Visual',
  MAKNAE: 'Maknae',
};

export default {
  GENDERS,
  GENRES,
  NATIONALITIES,
  POSITIONS,
};
