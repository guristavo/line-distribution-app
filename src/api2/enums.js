export const CATEGORY = {
  OFFICIAL: 'OFFICIAL',
  SHOULD: 'SHOULD',
  WOULD: 'WOULD',
};

export const GENDERS = {
  FEMALE: 'FEMALE',
  MALE: 'MALE',
  UNKNOWN: 'UNKNOWN',
};

export const GENRES = {
  CPOP: 'C-POP',
  JPOP: 'J-POP',
  KPOP: 'K-POP',
  'C-POP': 'C-POP',
  'J-POP': 'J-POP',
  'K-POP': 'K-POP',
  OTHER: 'OTHER',
  POP: 'POP',
};

export const NATIONALITIES = {
  AMERICAN: 'AMERICAN',
  BRAZILIAN: 'BRAZILIAN',
  BRITISH: 'BrBRITISHitish',
  CANADIAN: 'CANADIAN',
  CHINESE: 'CHINESE',
  INDIAN: 'INDIAN',
  INDONESIAN: 'INDONESIAN',
  FILIPINO: 'FILIPINO',
  JAPANESE: 'JAPANESE',
  KOREAN: 'KOREAN',
  OTHER: 'OTHER',
  TAIWANESE: 'TAIWANESE',
  THAI: 'THAI',
  VIETNAMISE: 'VIETNAMISE',
};

export const POSITIONS = {
  LEADER: 'LEADER',
  MAIN_VOCALIST: 'MAIN_VOCALIST',
  MAIN_RAPPER: 'MAIN_RAPPER',
  MAIN_DANCER: 'MAIN_DANCER',
  LEAD_VOCALIST: 'LEAD_VOCALIST',
  LEAD_RAPPER: 'LEAD_RAPPER',
  LEAD_DANCER: 'LEAD_DANCER',
  VOCALIST: 'VOCALIST',
  RAPPER: 'RAPPER',
  DANCER: 'DANCER',
  CENTER: 'CENTER',
  FACE: 'FACE',
  VISUAL: 'VISUAL',
  MAKNAE: 'MAKNAE',
};

export const UNKNOWN = 'UNKNOWN';

const getEnum = (value = '', library = '') => {
  let result;

  if (value === UNKNOWN) return UNKNOWN;

  switch (library) {
    case 'CATEGORY':
      result = CATEGORY[value];
      break;
    case 'GENDER':
      result = GENDERS[value];
      break;
    case 'GENRE':
      result = GENRES[value];
      break;
    case 'NATIONALITY':
      result = NATIONALITIES[value];
      break;
    case 'POSITIONS':
      result = POSITIONS[value];
      break;
    default:
      throw new Error(`Invalid enum library: ${library}`);
  }

  if (result) {
    return result;
  }
  return UNKNOWN;
};

export default getEnum;
