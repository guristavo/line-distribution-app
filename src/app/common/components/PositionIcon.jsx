import React from 'react';
import PropTypes from 'prop-types';

// Import position icons
import iconLeader from '../../../images/icon-leader.svg';
import iconMainVocalist from '../../../images/icon-main-vocalist.svg';
import iconMainRapper from '../../../images/icon-main-rapper.svg';
import iconMainDancer from '../../../images/icon-main-dancer.svg';
import iconLeadVocalist from '../../../images/icon-lead-vocalist.svg';
import iconLeadRapper from '../../../images/icon-lead-rapper.svg';
import iconLeadDancer from '../../../images/icon-lead-dancer.svg';
import iconVocalist from '../../../images/icon-vocalist.svg';
import iconDancer from '../../../images/icon-dancer.svg';
import iconRapper from '../../../images/icon-rapper.svg';
import iconCenter from '../../../images/icon-center.svg';
import iconVisual from '../../../images/icon-visual.svg';
import iconMaknae from '../../../images/icon-maknae.svg';
import iconAll from '../../../images/icon-all.svg';

const PositionIcon = ({ className, position }) => {
  const iconBank = {
    LEADER: iconLeader,
    MAIN_VOCALIST: iconMainVocalist,
    MAIN_RAPPER: iconMainRapper,
    MAIN_DANCER: iconMainDancer,
    LEAD_VOCALIST: iconLeadVocalist,
    LEAD_RAPPER: iconLeadRapper,
    LEAD_DANCER: iconLeadDancer,
    VOCALIST: iconVocalist,
    RAPPER: iconDancer,
    DANCER: iconRapper,
    CENTER: iconCenter,
    VISUAL: iconVisual,
    MAKNAE: iconMaknae,
    ALL: iconAll,
  };

  return (
    <span className={className}>
      <img
        className="position-icon"
        src={iconBank[position]}
        alt={`Position: ${position}`}
        title={position}
      />
    </span>
  );
};

PositionIcon.propTypes = {
  className: PropTypes.string,
  position: PropTypes.string.isRequired,
};

PositionIcon.defaultProps = {
  className: 'position-icon-container--inline',
};

export default PositionIcon;
