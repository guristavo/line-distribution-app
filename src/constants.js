import ARTISTS_DB from './database';

export const COLORS_LIST = ['blood', 'redOrange', 'peach', 'brown', 'orange', 'dirt', 'lightBrown', 'sand', 'gold', 'tan', 'yellow', 'olive', 'pee', 'yellowGreen', 'green', 'darkGreen', 'forest', 'lime', 'teal', 'cyan', 'turquoise', 'navy', 'blue', 'purple', 'violet', 'rose', 'plum', 'pink', 'redViolet', 'hotPink', 'wine', 'salmon', 'red', 'lightGrey', 'grey', 'darkGrey'];

export const KEYS = {
    49: {
        keyCode: 49,
        keyFace: '1',
        id: 0
    },
    50: {
        keyCode: 50,
        keyFace: '2',
        id: 1
    },
    51: {
        keyCode: 51,
        keyFace: '3',
        id: 2
    },
    52: {
        keyCode: 52,
        keyFace: '4',
        id: 3
    },
    53: {
        keyCode: 53,
        keyFace: '5',
        id: 4
    },
    54: {
        keyCode: 54,
        keyFace: '6',
        id: 5
    },
    55: {
        keyCode: 55,
        keyFace: '7',
        id: 6
    },
    56: {
        keyCode: 56,
        keyFace: '8',
        id: 7
    },
    57: {
        keyCode: 57,
        keyFace: '9',
        id: 8
    },
    48: {
        keyCode: 48,
        keyFace: '0',
        id: 9
    },
    81: {
        keyCode: 81,
        keyFace: 'Q',
        id: 10
    },
    87: {
        keyCode: 87,
        keyFace: 'W',
        id: 11
    },
    69: {
        keyCode: 69,
        keyFace: 'E',
        id: 12
    },
    82: {
        keyCode: 82,
        keyFace: 'R',
        id: 13
    },
    84: {
        keyCode: 84,
        keyFace: 'T',
        id: 14
    },
    89: {
        keyCode: 89,
        keyFace: 'Y',
        id: 15
    },
    85: {
        keyCode: 85,
        keyFace: 'U',
        id: 16
    },
    73: {
        keyCode: 73,
        keyFace: 'I',
        id: 17
    },
    79: {
        keyCode: 79,
        keyFace: 'O',
        id: 18
    },
    80: {
        keyCode: 80,
        keyFace: 'P',
        id: 19
    },
    65: {
        keyCode: 65,
        keyFace: 'A',
        id: 20
    },
    83: {
        keyCode: 83,
        keyFace: 'S',
        id: 21
    },
    68: {
        keyCode: 68,
        keyFace: 'D',
        id: 22
    },
    70: {
        keyCode: 70,
        keyFace: 'F',
        id: 23
    },
    71: {
        keyCode: 71,
        keyFace: 'G',
        id: 24
    },
};

export const KEYCODE_LIST = [49, 50, 51, 52, 53, 54, 55, 56, 57, 48, 81, 87, 69, 82, 84, 89, 85, 73, 79, 80, 65, 83, 68, 70, 71];

export const KEY_LIST = [1, 2, 3, 4, 5, 6, 7, 8, 9, 0, 'Q', 'W', 'E', 'R', 'T', 'Y', 'U', 'I', 'O', 'P', 'A', 'S', 'D', 'F', 'G'];

export const ARTISTS = ARTISTS_DB;

export const ARTITST_PLACEHOLDER = {
    bandName: 'No Artists Selected',
    members: [''],
    colors: ['']
};
