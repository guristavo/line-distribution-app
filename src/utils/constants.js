const COLORS = {
  col000001: 'red',
  col000002: 'blood',
  col000003: 'coral',
  col000004: 'peach',
  col000005: 'brown',
  col000006: 'orange',
  col000007: 'gold',
  col000008: 'yellow',
  col000009: 'olive',
  col000010: 'lemon',
  col000011: 'lime',
  col000012: 'green',
  col000013: 'basil',
  col000014: 'forest',
  col000015: 'jade',
  col000016: 'teal',
  col000017: 'cyan',
  col000018: 'turquoise',
  col000019: 'navy',
  col000020: 'blue',
  col000021: 'sky',
  col000022: 'purple',
  col000023: 'violet',
  col000024: 'rose',
  col000025: 'mauve',
  col000026: 'magenta',
  col000027: 'jam',
  col000028: 'pink',
  col000029: 'burgundy',
  col000030: 'salmon',
};

const DEFAULT_COLORS = {
  default: '#000000',
  black: '#000000',
  blue: '#598ffd',
  gray: '#707070',
  green: '#49d156',
  orange: '#fea245',
  pink: '#fb9cf0',
  purple: '#9e8ef0',
  red: '#d14949',
  teal: '#34ac9e',
  white: 'FFFFFF',
  yellow: '#ffd700',
};

const FLAGS_LIST = {
  AMERICAN: 'usa',
  BRAZILIAN: 'brazil',
  BRITISH: 'uk',
  CANADIAN: 'canada',
  CHINESE: 'china',
  JAPANESE: 'japan',
  KOREAN: 'korea',
  PLANTIAN: 'planted',
  TAIWANESE: 'taiwan',
  THAI: 'thailand',
  VIETNAMISE: 'vietnam',
};

const GENRES_DB = {
  'C-Pop': 'CPOP',
  'J-Pop': 'JPOP',
  'K-Pop': 'KPOP',
  Other: 'OTHER',
  Pop: 'POP',
};

const POSITIONS_LIST = [
  'MAIN_VOCALIST',
  'MAIN_RAPPER',
  'MAIN_DANCER',
  'LEAD_VOCALIST',
  'LEAD_RAPPER',
  'LEAD_DANCER',
  'VOCALIST',
  'RAPPER',
  'DANCER',
  'LEADER',
  'CENTER',
  'VISUAL',
];

const PROFILE_PICTURE_URL = '/images/profiles/';

const GROUPS_PICTURE_URL = '/images/groups/';

const SYNC_KEY_LIST = {
  1: true,
  2: true,
  3: true,
  4: true,
  5: true,
};

const SYNC_KEY_COLOR_LIST = {
  1: '1',
  2: '8',
  3: '12',
  4: '20',
  5: '26',
};

const YOUTUBE_URLS = ['cudssvDuOpc'];

export default {
  COLORS,
  DEFAULT_COLORS,
  FLAGS_LIST,
  GENRES_DB,
  GROUPS_PICTURE_URL,
  POSITIONS_LIST,
  PROFILE_PICTURE_URL,
  SYNC_KEY_LIST,
  SYNC_KEY_COLOR_LIST,
  YOUTUBE_URLS,
};
