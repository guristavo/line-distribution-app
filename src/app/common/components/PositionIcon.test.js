import React from 'react';
import { render, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import PositionIcon from './PositionIcon';

import constants from '../../../utils/constants';
import enums from '../../../utils/readable-enums';

describe('common/PositionIcon', () => {
  afterEach(cleanup);

  it('renders', () => {
    const { asFragment } = render(<PositionIcon position="LEADER" />);
    expect(asFragment()).toMatchSnapshot();
  });

  const listOfPOsitions = Object.entries(constants.POSITIONS_LIST_ICON);

  for (let i = 0; i < listOfPOsitions.length; i++) {
    const [position, classString] = listOfPOsitions[i];
    it(`renders the correct position icon: ${position}`, () => {
      const { getByTestId } = render(<PositionIcon position={position} />);

      const element = getByTestId('position-icon');
      const classList = element.classList.value;
      expect(element).toBeInTheDocument();
      expect(classList.includes('position-icon')).toBe(true);
      expect(classList.includes(`position-icon--${classString}`)).toBe(true);
    });
  }

  for (let i = 0; i < listOfPOsitions.length; i++) {
    const [position, classString] = listOfPOsitions[i];
    it(`renders the correct position icon with display name: ${position}`, () => {
      const { getByTestId, getByText } = render(
        <PositionIcon position={position} displayName />
      );

      const container = getByTestId('position-icon-container');
      const classList = container.classList.value;
      expect(container).toBeInTheDocument();
      expect(classList.includes('position-icon-container')).toBe(true);

      const element = getByTestId('position-icon');
      const classListEl = element.classList.value;
      expect(element).toBeInTheDocument();
      expect(classListEl.includes('position-icon')).toBe(true);
      expect(classListEl.includes(`position-icon--${classString}`)).toBe(true);

      const text = getByText(enums.POSITIONS[position]);
      expect(text).toBeInTheDocument();
    });
  }

  it('renders the unknown icon when position is not in the positions list constant', () => {
    const { getByTestId, getByText } = render(
      <PositionIcon position="BOLA" displayName />
    );

    const container = getByTestId('position-icon-container');
    const classList = container.classList.value;
    expect(container).toBeInTheDocument();
    expect(classList.includes('position-icon-container')).toBe(true);

    const element = getByTestId('position-icon');
    const classListEl = element.classList.value;
    expect(element).toBeInTheDocument();
    expect(classListEl.includes('position-icon')).toBe(true);
    expect(classListEl.includes('position-icon--unknown')).toBe(true);

    const text = getByText('BOLA');
    expect(text).toBeInTheDocument();
  });

  it('renders a display name when a string is passed as the displayName argument', () => {
    const { getByTestId, getByText } = render(
      <PositionIcon position="LEADER" displayName="Custom Display Name" />
    );

    const container = getByTestId('position-icon-container');
    const classList = container.classList.value;
    expect(container).toBeInTheDocument();
    expect(classList.includes('position-icon-container')).toBe(true);

    const element = getByTestId('position-icon');
    const classListEl = element.classList.value;
    expect(element).toBeInTheDocument();
    expect(classListEl.includes('position-icon')).toBe(true);
    expect(classListEl.includes('position-icon--leader')).toBe(true);

    const text = getByText('Custom Display Name');
    expect(text).toBeInTheDocument();
  });
});
