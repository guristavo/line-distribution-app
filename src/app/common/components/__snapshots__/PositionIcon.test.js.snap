// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`common/PositionIcon renders 1`] = `
<DocumentFragment>
  <span
    class="position-icon position-icon--leader "
    data-testid="position-icon"
    title="Position: Leader"
  />
</DocumentFragment>
`;
